src/main/train.cpp

```diff
@@ -495,6 +495,55 @@ void printUsage(char *argv[], Config config) {
     // [[[end]]]
 }
 
+int setDefaults(Config& config, string dataset) {
+    string dataDir;
+    string trainFile;
+    string validateFile;
+    int loadOnDemand = 0;
+
+    if(dataset == "mnist" || dataset.empty()) {
```

Comment:
It shouldn't check dataset.empty() like this.
I see that you did this to set the default values in `Config`, but when dataset isn't used this data isn't needed, thus this doesn't need to be set.